Sources/SourceControl/RepositoryManager.swift

```diff
@@ -384,6 +384,8 @@ public class RepositoryManager {
             } catch {
                 // Fetch without populating the cache in the case of an error.
                 print("Skipping cache due to an error: \(error)")
+                // It is possible that we already created the directory before failing, so clear leftover data if present.
+                try fileSystem.removeFileTree(repositoryPath)
```

Comment:
I'm pretty sure that this can't spuriously fail because `repositoryPath` is missing, but could it fail because its parent path is missing or other reason?  i.e. would it be better to deal with exceptions from here or let them raise?

Reply:
testing if it exists before deleting sounds reasonable, then raising if we can't delete otherwise we will have downstream issues?

Reply:
It's probably not a problem in this particular case, but in general, checking for existence first can be prone to race conditions.  So a pattern that usually works well for cases like this is to just do the file system operation but to silently ignore benign problems.  In this case, an ENOENT error is fine but anything else should be reported.  That might be what the underlying API already does, so my question was just to confirm that semantic.  I'll check the code.

Reply:
Today it does this: 
```if self.exists(path, followSymlink: false) {
    try FileManager.default.removeItem(atPath: path.pathString)
 }
```

So the race potential is there.

Reply:
That's possibly a follow-on fix down in TSC, but it looks as if the code in this particular diff is correct.  I think it's right that `try fileSystem.removeFileTree(repositoryPath)` doesn't throw an error if the file isn't already there (much like `mkdir` doesn't complain if the directory _is_ there).  So the code in this PR is right but TSC could use a fix, it looks like.  So I have no concerns about this PR.  Thanks!

Reply:
I think what would make the most sense is to leave this as-is but change `removeFileTree` in TSC to not throw on `ENOENT` as suggested, but for the 5.4 cherry-pick we inline the `ENOENT` check to reduce the surface area of the change.